### PR TITLE
Fix MC-123739

### PR DESCRIPTION
**Pull request by aws404**
## Description
This PR will fix MC-123739: Recipe book entries are not sorted in any meaningful manner. In fact, the order can even change randomly after a reload. 
Recipie book entries are now sorted by the ID of the first recipe in the group, leading to a consistant order in the GUI.

### CODE REVIEW

File: PATCHED.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 | [MC-176559](https://bugs.mojang.com/browse/MC-176559) | Breaking process resets when a pickaxe enchanted with Mending mends by XP / Mending slows down breaking blocks again *(fabric only)* |
 | [MC-231097](https://bugs.mojang.com/browse/MC-231097) | Holding the "Use" button continues to slow down the player even after the used item has been dropped                                 |
 | [MC-234898](https://bugs.mojang.com/browse/MC-234898) | The "Get a trial!" button isn't consistently displayed within the realms menu                                                        |
+| [MC-123739](https://bugs.mojang.com/browse/MC-123739) | Recipe book entries are not sorted in any meaningful manner. In fact, the order can even change randomly after a reload.             |
```
Comment (isXander) [inline PATCHED.md:21]: please add this in alphabetical order per bug id

File: common/src/main/java/cc/woverflow/debugify/mixins/client/mc123739/ClientRecipeBookMixin.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+public abstract class ClientRecipeBookMixin {
+
+    /**
+     * Method: reload
+     * Anonymous Class: map.forEach((recipeBookGroup, list) -> {
+     *     # HERE #
+     * })
+     */
+    @SuppressWarnings("UnresolvedMixinReference")
+    @Redirect(method = "method_30279", at = @At(value = "INVOKE", target = "Ljava/util/stream/Stream;map(Ljava/util/function/Function;)Ljava/util/stream/Stream;"))
```
Comment (isXander) [inline common/src/main/java/cc/woverflow/debugify/mixins/client/mc123739/ClientRecipeBookMixin.java:25]: have you tested this in forge? I had issues with lambdas in forge being remapped to mojang srgs with architectury. [see this file](https://github.com/W-OVERFLOW/Debugify/blob/1.18/common/src/main/java/cc/woverflow/debugify/mixins/client/mc234898/RealmsMainScreenMixin.java#L40)

**Review by isXander — CHANGES_REQUESTED** [review]

File: common/src/main/java/cc/woverflow/debugify/mixins/client/mc123739/ClientRecipeBookMixin.java (hunk 2 above)
Comment (aws404) [reply common/src/main/java/cc/woverflow/debugify/mixins/client/mc123739/ClientRecipeBookMixin.java:25]: Its working for both fabric and forge, in both Dev and Prod environements.

**Comment by isXander** [comment]
Did you mean to draft this?

**Comment by aws404** [comment]
> Did you mean to draft this?

Yep, I have changed the way I went about this to reference the reload method directly rather than the lambda.

**Pull request merged**